### PR TITLE
Add format check to ci

### PR DESCRIPTION
**Pull request by nathanaelhoun**
## Changes

- [x] Updated CI to add Prettier check
- [x] Better workflow organization for more clarity
- [x] Remove workflow runs on pushes to `main` as we use PRs all the time

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -31,7 +31,8 @@
     "eject": "react-scripts eject",
     "lint": "eslint src",
     "lint:fix": "eslint --fix src",
-    "format": "prettier --write ."
+    "format": "prettier --write .",
+    "format:check": "prettier --check ."
```
Comment (valentinperignon) [inline client/package.json:35]: Maybe we could keep the same format as for "lint".
That is to say that "format" checks and "format:write" corrects (write)

File: server/package.json [hunk 2]
```diff
@@ -9,7 +9,8 @@
     "start": "node .",
     "lint": "eslint .",
     "lint:fix": "eslint --fix .",
-    "format": "prettier --write ."
+    "format": "prettier --write .",
```
Comment (valentinperignon) [inline server/package.json:12]: Same :)

**Review by valentinperignon — APPROVED** [review]
LGTM, just one comment

**Review by FrancoisPog — APPROVED** [review]
LGTM !

File: client/package.json (hunk 1 above)
Comment (nathanaelhoun) [reply client/package.json]: You are right!

**Review by valentinperignon — APPROVED** [review]
LGTM!!

**Pull request merged**